### ISSUE TITLE
Impossible to annotate parameter properties in an isolatedDeclarations compatible way

### ISSUE THREAD

**Issue by blickly** [microsoft/TypeScript#60123]
### 🔎 Search Terms

isolatedDeclarations, parameter properties, implicit undefined

### 🕗 Version & Regression Information

- Present since `isolatedDeclarations` was added in TS5.5

### ⏯ Playground Link

https://www.typescriptlang.org/play/?downlevelIteration=true&importHelpers=true&target=99&module=1&isolatedDeclarations=true&noCheck=true&exactOptionalPropertyTypes=true&ssl=3&ssc=2&pln=1&pc=1#code/KYDwDg9gTgLgBAYwDYEMDOa4CEVTgbwCg4TEIA7NGKAVwRmgAowoBLANxRmDhAH4AXHACCUKCgCecAD5wa5ACbAAZq3LAFASgIBfQjqA

### 💻 Code

```ts
export class Bar {
    constructor(private x?: Array | undefined) {}
}
```

### 🙁 Actual behavior

Error:
> Declaration emit for this parameter requires implicitly adding undefined to it's type. This is not supported with --isolatedDeclarations.

### 🙂 Expected behavior

Should not give an error since the undefined is already explicit.

**Comment by jcalz** [comment]
Why are you writing `Array` without its type argument like `Array<any>` or `Array<number>`? Isn't that the source of the problem? Or is it just a typo?

**Comment by Andarist** [comment]
That is the issue but I'd consider the raised error to be a bug, at the very least it's quite confusing here

**Comment by blickly** [comment]
Oops, you are right.  I oversimplified my original example to get something unrealistic.  My original repro case was actually:

```
import {Foo} from 'other';

export class Bar {
    constructor(private x?: Foo | undefined) {}
}
```

https://www.typescriptlang.org/play/?downlevelIteration=true&importHelpers=true&target=99&module=1&isolatedDeclarations=true&noCheck=true&exactOptionalPropertyTypes=true#code/JYWwDg9gTgLgBAbwGIQgXzgMyhEcDkEMAFgKZT4DcAUNaQB6SxwDGANgIYDOXcAQhyiJqcUawgA7LjCgBXFjGgAKMFGAA3DjFJx6AfgBccFBDgAfOLIkATUpmATS1gJSI01NEA

But I think that the fix will work either way.
